### PR TITLE
boards: Add stm32f407g-disc1 Discovery devboard support

### PR DESCRIPTION
**Pull request by finikorg**
Add support for stm32f407g-disc1 board, this is mofified version of
stm32f4_disco board.

Fixes: #8810

Signed-off-by: Andrei Emeltchenko <andrei.emeltchenko@intel.com>

### CODE REVIEW

File: boards/arm/stm32f407g_disc1/Kconfig.board [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# Kconfig - STM32F4DISCOVERY board configuration
```
Comment (ydamigos) [inline boards/arm/stm32f407g_disc1/Kconfig.board]: STM32F407G-DISC1?

File: boards/arm/stm32f407g_disc1/Kconfig.board [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+#
+# Copyright (c) 2016 Linaro Limited.
```
Comment (ydamigos) [inline boards/arm/stm32f407g_disc1/Kconfig.board]: Shouldn't this be 2018, Intel Corporation?

File: boards/arm/stm32f407g_disc1/Kconfig.defconfig [hunk 3]
```diff
@@ -0,0 +1,27 @@
+# Kconfig - STM32F4DISCOVERY board configuration
```
Comment (ydamigos) [inline boards/arm/stm32f407g_disc1/Kconfig.defconfig]: STM32F407G-DISC1?

File: boards/arm/stm32f407g_disc1/Kconfig.defconfig [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+#
+# Copyright (c) 2016 Linaro Limited.
```
Comment (ydamigos) [inline boards/arm/stm32f407g_disc1/Kconfig.defconfig]: 2018, Intel Corporation?

File: boards/arm/stm32f407g_disc1/board.h [hunk 5]
```diff
@@ -0,0 +1,12 @@
+/*
+ * Copyright (c) 2016 Linaro Limited.
```
Comment (ydamigos) [inline boards/arm/stm32f407g_disc1/board.h]: 2018, Intel Corporation?

File: boards/arm/stm32f407g_disc1/pinmux.c [hunk 6]
```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) 2016 Linaro Limited.
```
Comment (ydamigos) [inline boards/arm/stm32f407g_disc1/pinmux.c]: 2018, Intel Corporation?

File: boards/arm/stm32f407g_disc1/stm32f407g_disc1.dts [hunk 7]
```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (c) 2017 Linaro Limited
```
Comment (ydamigos) [inline boards/arm/stm32f407g_disc1/stm32f407g_disc1.dts]: 2018, Intel Corporation?

File: boards/arm/stm32f407g_disc1/stm32f407g_disc1.yaml [hunk 8]
```diff
@@ -0,0 +1,9 @@
+identifier: stm32f407g_disc1
+name: STM32F407G-DISC1
+type: mcu
+arch: arm
+toolchain:
+  - zephyr
+  - gccarmemb
+supported:
+  - pwm
```
Comment (ydamigos) [inline boards/arm/stm32f407g_disc1/stm32f407g_disc1.yaml]: You could add gpio, pwm and usb_device.

**Review by ydamigos — CHANGES_REQUESTED** [review]
Looks good. Some minor comments regarding copyright and board name in comments.

File: boards/arm/stm32f407g_disc1/doc/stm32f407g_disc1.rst [hunk 9]
```diff
@@ -0,0 +1,202 @@
...
+- USB 2.0 OTG FS with on-chip PHY
+- USB 2.0 OTG HS/FS with dedicated DMA, on-chip full-speed PHY and ULPI
+- 10/100 Ethernet MAC with dedicated DMA
+- 8- to 14-bit parallel camera
+- CRC calculation unit
+- True random number generator
+- DMA Controller
+
+More information about STM32F407VG can be found here:
+       - `STM32F407VG on www.st.com`_
```
Comment (dbkinder) [inline boards/arm/stm32f407g_disc1/doc/stm32f407g_disc1.rst]: add a blank line before this bullet list, and don't indent the items:

    More information about STM32F407VG can be found here:

    - `STM32F407VG on www.st.com`_
    - `STM32F407 reference manual`_

File: boards/arm/stm32f407g_disc1/doc/stm32f407g_disc1.rst [hunk 10]
```diff
@@ -0,0 +1,202 @@
...
++-----------+------------+-------------------------------------+
+| PWM       | on-chip    | pwm                                 |
++-----------+------------+-------------------------------------+
+| USB       | on-chip    | USB device                          |
++-----------+------------+-------------------------------------+
+
+Other hardware features are not yet supported on Zephyr porting.
+
+The default configuration can be found in the defconfig file:
+
```
Comment (dbkinder) [inline boards/arm/stm32f407g_disc1/doc/stm32f407g_disc1.rst]: delete the blank line and the leading spaces before  the filename

    The default configuration can be found in the defconfig file:
    ``boards/arm/stm32f407g_disc1/stm32f407g_disc1_defconfig``

**Review by dbkinder — CHANGES_REQUESTED** [review]
looks good, just a couple of formatting fixes

**Comment by finikorg** [comment]
@ydamigos @dbkinder Updated PR.

**Review by ydamigos — APPROVED** [review]
Looks good.

**Comment by finikorg** [comment]
@dbkinder Could you recheck?

**Comment by erwango** [comment]
Thanks for this work, but I wonder if we should introduce different board revisions as new boards.
Ideal would be to support various board revisions for a board.
I raised an issue once (https://github.com/zephyrproject-rtos/zephyr/issues/4315), it might be good to do this instead

File: boards/arm/stm32f407g_disc1/stm32f407g_disc1_defconfig [hunk 11]
```diff
@@ -0,0 +1,42 @@
...
+CONFIG_CLOCK_STM32_PLL_M_DIVISOR=8
+CONFIG_CLOCK_STM32_PLL_N_MULTIPLIER=336
+CONFIG_CLOCK_STM32_PLL_P_DIVISOR=2
+CONFIG_CLOCK_STM32_PLL_Q_DIVISOR=7
+CONFIG_CLOCK_STM32_AHB_PRESCALER=1
+CONFIG_CLOCK_STM32_APB1_PRESCALER=4
+CONFIG_CLOCK_STM32_APB2_PRESCALER=2
+
+#enable pwm
+CONFIG_PWM=y
```
Comment (erwango) [inline boards/arm/stm32f407g_disc1/stm32f407g_disc1_defconfig]: Remove this line

**Review by erwango — CHANGES_REQUESTED** [review]
@finikorg , let's merge this PR has I have no other option to propose for now, and technically both boards revisions have their own reference

One comment though:
Remove board.h

**Comment by erwango** [comment]
@dbkinder , can you review again?

**Review by erwango — APPROVED** [review]

**Review by dbkinder — APPROVED** [review]
doc changes LGTM, thanks.

File: boards/arm/stm32f407g_disc1/stm32f407g_disc1_defconfig [hunk 12]
```diff
@@ -0,0 +1,39 @@
+CONFIG_ARM=y
+CONFIG_BOARD_STM32F407G_DISC1=y
+CONFIG_SOC_SERIES_STM32F4X=y
+CONFIG_SOC_STM32F407XG=y
+# 168MHz system clock
+CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC=168000000
+CONFIG_SYS_CLOCK_TICKS_PER_SEC=1000
+
+# Enable MPU
+CONFIG_STM32_ARM_MPU_ENABLE=y
```
Comment (nashif) [inline boards/arm/stm32f407g_disc1/stm32f407g_disc1_defconfig]: you need to remove this, it is not defined anywhere

**Review by nashif — CHANGES_REQUESTED** [review]

**Pull request closed without merging**